app/controllers/api/transformation_mappings_controller.rb

```diff
@@ -49,6 +49,15 @@ def vm_flavor_fit_resource(_type, _id, data)
       end
     end
 
+    def add_mapping_item_resource(type, id, data)
+      mapping = resource_search(id, type, collection_class(type))
+      mapping.transformation_mapping_items.append(create_mapping_items([data]))
+      mapping.save!
+      mapping
+    rescue StandardError => err
+      raise BadRequestError, "Failed to update Transformation Mapping - #{err}"
+    end
+
```

Comment:
A little syntactic sugar you can use here is the `Object#tap` method. So you can write that like this:

```
resource_search(id, type, collection_class(type)).tap do |mapping|
  mapping.transformation_mapping_items.append(create_mapping_items([data]))
  mapping.save!
end
```

Reply:
All right, fixed.